includes/Core/Authentication/Authentication.php

```diff
@@ -600,6 +611,55 @@ private function allowed_redirect_hosts( $hosts ) {
 		return $hosts;
 	}
 
+	/**
+	 * Gets related REST routes.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array List of REST_Route objects.
+	 */
+	private function get_rest_routes() {
+		$can_authenticate = function() {
+			return current_user_can( Permissions::AUTHENTICATE );
+		};
+
+		return array(
+			new REST_Route(
+				'core/user/data/authentication',
+				array(
+					array(
+						'methods'             => WP_REST_Server::READABLE,
+						'callback'            => function( WP_REST_Request $request ) {
+							$oauth_client = $this->get_oauth_client();
+							$access_token = $oauth_client->get_access_token();
+
+							$data = array(
+								'isAuthenticated' => ! empty( $access_token ),
+								'requiredScopes'  => $oauth_client->get_required_scopes(),
+								'grantedScopes'   => ! empty( $access_token ) ? $oauth_client->get_granted_scopes() : array(),
+							);
+
+							return new WP_REST_Response( $data );
+						},
+						'permission_callback' => $can_authenticate,
+					),
+				)
+			),
+			new REST_Route(
+				'core/user/data/disconnect',
+				array(
+					array(
+						'methods'             => WP_REST_Server::EDITABLE,
+						'callback'            => function( WP_REST_Request $request ) {
+							return new WP_REST_Response( $this->disconnect() );
```

Comment:
I realize this is how it was before but we probably shouldn't use `disconnect()` here as it has no return value.

We should instead call this above the response and then respond with `{ success: true }` since it seems like this method should always succeed.